library/std/src/lazy.rs

```diff
@@ -316,13 +316,7 @@ impl<T> SyncOnceCell<T> {
     /// ```
     #[unstable(feature = "once_cell", issue = "74465")]
     pub fn into_inner(mut self) -> Option<T> {
-        // SAFETY: Safe because we immediately free `self` without dropping
-        let inner = unsafe { self.take_inner() };
-
-        // Don't drop this `SyncOnceCell`. We just moved out one of the fields, but didn't set
-        // the state to uninitialized.
-        mem::forget(self);
-        inner
+        self.take()
```

Comment:
Since this is still unstable - is there any reason to have both these methods? I think just having `take()` would be more clear.

Reply:
`into_inner()` could make use of forgetting `self` for efficiency. But in this particular implementation, `.take()` followed by dropping it is just as efficient. So I think it's good to have `into_inner` separately. Also for consistency with the other cells and atomics.